yabt/buildcontext.py

```diff
@@ -181,6 +183,93 @@ def get_target_extraction_context(self, build_file_path: str) -> dict:
     #     """Register a named BuildEnv Docker image in this build context."""
     #     self.buildenv_images[name] = docker_image
 
+    def get_buildenv_graph(self):
```

Comment:
If I understand what's going on here, then in order to collect all the descendants, you collect all the subgraphs of the nodes in buildenvs. This is can be very non-optimal, as these graphs may have huge overlap,  which you'll discover only after the reduction. If you want to keep it this way, please add a TODO about it.

Reply:
the code is it is now collects all the subsets of nodes that all buildenvs depend on, and then gets the subgraph of the union of all subsets.

I agree this can be very non-optimal - I could not find a built-in way to do it better, and I didn't want to go into implementing my own algo for that, given that the concept of buildenvs might change soon, so I'll leave it this way with a TODO.